pr-builder.rst

```diff
@@ -0,0 +1,99 @@
+
+Google Summer of Code
+=====================
+
+Google Summer of Code is a global program where students work with an open source organization
+on a 3 month programming project. **Autobuild Documentation in Pull Requests** was one of the project ideas on
+Read the Docs `Project Ideas`_ for Google Summer of Code 2019. I applied for the project and got accepted.
+
+I have started working on this project from May 2019.
+All of my work on this project can be seen in the `Pull Request Builder Board`_.
+
+Background
+==========
+
+Many of our users wanted a way to visualize the documentation update that was made in a pull/merge request.
+They also want to know whether the documentation build will pass before merging the pull/merge request to master.
+This would allow users to have more confidence on the pull request
+and make the pull/merge request less likely to break the documentation after merging.
+So, to achieve this Read the Docs `core team`_ selected **Autobuild docs for Pull Requests**
+as one of the projects of Google Summer of Code 2019.
+
+Pull Request Builder Features
+=============================
+
+Currently I have implemented these features and working on more upcoming features.
+Some of the major features are as following:
+
+- **Creating External Versions:** We create an External Version when we receive pull_request opened/reopened
+  webhook event for a project from gitHub and trigger a build for that version.
+
+- **Synchronizing External Versions:** Whenever there is a new commit on the pull/merge request,
+  we synchronize the external version for that pull/merge request and trigger a new build with the latest changes.
+
+- **Deleting External Versions:** Whenever the pull/merge request is closed or merged,
+  we delete the external Version associated with that pull/merge request.
+
+- **Warning Banner for pull/merge request documentation:** While building documentation for pull/merge requests
+  we add a warning banner at the top of those documentations to let the users know that
+  this documentation was generated for pull/merge requests and is not the main documentation for the project.
+
+- **Sending Status Notification to Status API:** We send Build status reports to Status API of the provider (e.g. GitHub) of
+  the project. When a build is triggered for a pull/merge request we send `state: pending` notification with build URL
+  and after the build has finished we send `state: success` notification if the build succeeded without any error
+  or `state: failure` notification if the build failed with build URL.
+  By going to the build URL provided in the status report users can view the build steps
+  and also see the documentation generated by that build.
+
+**Currently, we only support GitHub and planning to extend to GitLab and BitBucket**
+
+Getting Started
+===============
+
+**Autobuild Documentation in Pull Requests** is currently on Beta testing and only supports Github repositories.
```

Comment:
```suggestion
Build documentation in pull requests is currently on Beta testing and only supports Github repositories.
```

The formal name of the project is probably not *that* relevant to readers of the blog.